src/utils.ts

```diff
@@ -173,6 +193,35 @@ export async function resolveLabel(label: string, config: Config): Promise<strin
   return config.provider.resolveName(`${label}.${config.registrar.domain}`);
 }
 
+export async function lookupAddress(address: string, did: string | null, config: Config): Promise<Lookup>  {
+  // First get eventual ENS Registration
+  let label: string | null = null;
+  try {
+    label = await config.provider.lookupAddress(address);
+  } catch (error) {
+    console.log("LookupAddress: Not able to get ENS Name, trying with IDX");
+  }
+  if (label && await config.provider.resolveName(label)) {
+    return {type: "ens", content: await getRegistration(label, config) as Registration };
+  }
+  // Else lookup IDX profile
+  if (did && isDid(did)) {
+    const content = await resolveIdxProfile(did, config);
+    if (content) return { type: "idx", content };
+  }
+  // If everything fails return address
+  return { type: "address", content: { name: address }};
+}
+
+// Resolves an existant idx profile
+export async function resolveIdxProfile(did: string, config: Config): Promise<radProfile | null> {
+  return config.idx.api.get<radProfile>("radProfile", did);
```

Comment:
Is "radProfile" a string we choose when we create the document type? Are there not existing document types for people who already have an IDX?

Reply:
The `radProfile` string we chose when we define it in `config.json` as the definition, which we use in the initialization of the IDX instance.
```js
// config.json
"idx": {
  "definition": "kjzl6cwe1jw14b4i43fpebvxonfyu8ei1f1a3mg32xhlmeugddnsa943dvgkzly"
}
// config.ts
const idx = new IDX({ ceramic,
  aliases: {
    radProfile: config.idx.definition
  }
});
```
We could eventually use the [`basicProfile` definition](https://developers.idx.xyz/guides/definitions/default/#basic-profile) but it does not correlate with the information that gets saved on the ENS record.
Eventually when we allow users to save their profiles through the app, we could allow users to port their `basicProfile` definition to a `radProfile`

Reply:
Hmm I see. Perhaps it's worth supporting basicProfile first though, and seeing if there's a way of extending it?

Reply:
008cd890ab5a0b7c2b668509bec1fbbc098af819 solves this